### PR TITLE
Feat/add server context to routes

### PR DESCRIPTION
**Pull request by aburd**
A biggish refactor to adding Context to server requests. 

TL;DR
Most of this stuff is probably not really important, but how to access the DB is easier:
[You can see an example here](https://github.com/tokyomovie/tokyomovie/pull/23/files#diff-6e3404c5da8f047b80ad23cc30a21a8b67b577f946f0b93f9c2d3b9c784c01b7R17)

- Adds Context so that requests can have access to various utilities easily
  - Adds session info to this context for session creation (this is rarely used, so not really important)
  - Adds db, so db can be used on every request without import
    - You can now get a reference to the `db` in your routes on the request `ctx`, please see [here](https://github.com/tokyomovie/tokyomovie/pull/23/files#diff-6e3404c5da8f047b80ad23cc30a21a8b67b577f946f0b93f9c2d3b9c784c01b7R17) for an example
  - As an extra note, this allows us to do more stuff on server initialization...not sure what that would be, but yeah, it's just cleaner 
- Adds Dockerfile - just because I'll probably use this for deployment

## More info

https://fresh.deno.dev/docs/concepts/deployment#-docker
https://fresh.deno.dev/docs/examples/init-the-server

### CODE REVIEW

**Comment by aburd** [comment]
Mergin' everyone, review not important, but just take a look at the description
